src/mqt/ddsim/qasmsimulator.py

```diff
@@ -73,22 +74,57 @@ def target(self):
     def max_circuits(self):
         return None
 
-    def run(self, quantum_circuits: QuantumCircuit | list[QuantumCircuit], **options: dict[str, Any]) -> DDSIMJob:
+    @staticmethod
+    def _bind_parameters(
+        quantum_circuits: list[QuantumCircuit], parameter_values: Sequence[Sequence[float]] | None = None
+    ) -> list[QuantumCircuit]:
+        if parameter_values is None:
+            parameter_values = []
+
+        bound_circuits = [
+            qc.bind_parameters(dict(zip(qc.parameters, values)))
+            for qc, values in zip(quantum_circuits, parameter_values)
+        ]
+
+        if len(parameter_values) < len(quantum_circuits):
+            bound_circuits.extend(quantum_circuits[len(parameter_values) : len(quantum_circuits)])
```

Comment:
I would believe that this check can be removed once if the above recommended change is adopted.
```suggestion
```

Reply:
I removed this conditional in the new commit.